dyninstAPI/src/image.h

```diff
@@ -361,8 +356,9 @@ class image : public codeRange {
    ParseAPI::CodeObject *codeObject() const { return obj_; }
 
    bool isDyninstRTLib() const { return is_libdyninstRT; }
-   bool isAOut() const { return is_a_out; }
-   bool isSharedObj() const { 
+   bool isExecutable() const { return getObject()->isExec(); }
+   bool isSharedLibrary() const { return getObject()->isSharedLibrary(); }
+   bool isSharedObject() const { 
```

Comment:
What's the difference between `isSharedLibrary` and `isSharedObject`?

Reply:
isSharedObject is elf type if DYN.  It is only used to determine a startup_printf message.   renamed old function (was isSharedObj).

isSharedLibrary besides the elf type being DYN has to meet all the other requirement of being a shared library as determined by the Object class.